doc/source/upgrading_guide_upstream.rst

```diff
@@ -0,0 +1,408 @@
+   
+   .. include:: .res/text/important.package.removing-without-dependency.txt
+
+.. rubric:: Installing new packages
+
+#. Install the ``percona-server-server-57`` package
+
+   .. code-block:: bash
+
+      $ yum install Percona-Server-server-57 
+
+   If you intend to use |Percona Server| 5.7 with |TokuDB|, run :program:`yum
+   install` specifying the |TokuDB| package along with
+   ``percona-server-server-57``.
+
+   .. code-block:: bash
+
+      $ yum install percona-server-server-57 percona-server-tokudb-57
+
+#. Modify your :file:`my.cnf` configuration file and reinstall the plugins if necessary.
+
+.. _percona-server.in-place-upgrade.starting.upstream:
+
+Starting the server
+--------------------------------------------------------------------------------
+
+With |Percona Server| installed, start the :program:`mysql` service:
+
+.. code-block:: bash
+
+   $ service mysql start
+
+|red-hat-enterprise| and |centos| 7 automatically back up the previous
+configuration file to :file:`/etc/my.cnf.rpmsave` and install the default
+:file:`my.cnf`. After the upgrading or installation process completes, you can
+move the old configuration file back with all unsupported system removed or
+commented out.
+
+.. important::
+
+   If you use the |TokuDB| storage engine, comment out all variables specific to
+   |TokuDB| in your configuration files before starting the server. Otherwise,
+   the server will not be able to start.
+
+.. _percona-server.mysql-upgrade.run-manually:
+
+Running ``mysql_upgrade`` manually
+--------------------------------------------------------------------------------
+
+When the :program:`mysql` service is started, use :program:`mysql_upgrade` to
+migrate to the new grant tables. It will rebuild the indexes and do the
+necessary modifications.
+
+The installation script does *NOT* run :command:`mysql_upgrade` automatically as
+it was the case in previous versions. You need to run this command manually and
+then restart the ``mysql`` service.
+   
+|tip.run-all.root|
+
+#. Run :program:`mysql_upgrade`
+
+   .. code-block:: bash
+
+      $ mysql_upgrade
+
+   .. admonition:: Output Example
+
+      .. code-block:: guess
+
+	 Checking if update is needed.
+	 Checking server version.
+	 Running queries to upgrade MySQL server.
+	 Checking system database.
+	 mysql.columns_priv                                 OK
+	 mysql.db                                           OK
+	 mysql.engine_cost                                  OK
+	 ...
+	 Upgrade process completed successfully.
+	 Checking if update is needed.
+
+#. Restart the :program:`mysql` service.
+
+   .. code-block:: bash
+
+      $ service mysql restart
+
+After the service successfully restarts, you can use the new |Percona Server|
+5.7.
+
+.. important::
+
+   If you are using the |TokuDB| storage engine, re-enable the storage engine
+   plugin before running :program:`mysql_upgrade`. Otherwise, you'll get errors.
+
```

Comment:
You can't re-enable the TokuDB engine if it hasn't been installed with the upstream packages.